pkg/httpclient/stream.go

```diff
@@ -34,7 +33,7 @@ func Stream(res *http.Response, msgFactory func() interface{}, outputCh interfac
 		r := bufio.NewReader(res.Body)
 		dec := sse.NewDecoder(r)
 		for {
-			msg := msgFactory()
+			msg := reflect.New(msgType).Interface()
```

Comment:
If you save the value instead of throwing it away, there's no need for the `ValueOf` call below.